cli/cmd/root.go

```diff
@@ -314,13 +318,39 @@ func (options *proxyConfigOptions) validate() error {
 
 	if options.proxyCPURequest != "" {
 		if _, err := k8sResource.ParseQuantity(options.proxyCPURequest); err != nil {
-			return fmt.Errorf("Invalid cpu request '%s' for --proxy-cpu flag", options.proxyCPURequest)
+			return fmt.Errorf("Invalid cpu request '%s' for --proxy-cpu-request flag", options.proxyCPURequest)
 		}
 	}
 
 	if options.proxyMemoryRequest != "" {
 		if _, err := k8sResource.ParseQuantity(options.proxyMemoryRequest); err != nil {
-			return fmt.Errorf("Invalid memory request '%s' for --proxy-memory flag", options.proxyMemoryRequest)
+			return fmt.Errorf("Invalid memory request '%s' for --proxy-memory-request flag", options.proxyMemoryRequest)
+		}
+	}
+
+	if options.proxyCPULimit != "" {
+		cpuLimit, err := k8sResource.ParseQuantity(options.proxyCPULimit)
+		if err != nil {
+			return fmt.Errorf("Invalid cpu limit '%s' for --proxy-cpu-limit flag", options.proxyCPULimit)
+		}
+		if options.proxyCPURequest != "" {
+			// Not checking for error because option proxyCPURequest was already validated
+			if cpuRequest, _ := k8sResource.ParseQuantity(options.proxyCPURequest); cpuRequest.Value() > cpuLimit.Value() {
```

Comment:
Unfortunately `.Value()` doesn't catch problems here because it rounds values like `100m` to `1`. So something like this won't be caught:
` inject --proxy-cpu-request 200m --proxy-cpu-limit 100m`

You want to use `.MilliValue()` instead for doing the comparison.

Reply:
Oops, my bad.

Fixed.